sdk/go/x/auto/stack.go

```diff
@@ -471,22 +451,17 @@ func (s *Stack) Destroy(ctx context.Context, opts ...optdestroy.Option) (Destroy
 
 // Outputs get the current set of Stack outputs from the last Stack.Up().
 func (s *Stack) Outputs(ctx context.Context) (OutputMap, error) {
-	err := s.Workspace().SelectStack(ctx, s.Name())
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to get stack outputs")
-	}
-
 	// standard outputs
 	outStdout, outStderr, code, err := s.runPulumiCmdSync(ctx, nil, /* additionalOutputs */
-		"stack", "output", "--json",
+		"stack", "output", "--json", "--stack", s.Name(),
```

Comment:
I think this applies in all languages, but we could probably set `--stack` consistently from within `stack.runPulumiCmdSync`. Then it would be enforced for every command run on the stack including future functionality that gets added. 

https://github.com/pulumi/pulumi/blob/bcd14d806046a562218e074d1f7872ec6ecf2409/sdk/go/x/auto/stack.go#L742

I believe that this should work. Every stack command should require this flag. Those commands that don't likely belong on workspace instead of stack (pulumi version for instance).

Reply:
Ah yes, good idea :)